cli/src/commands/shell/index.ts

```diff
@@ -1,11 +1,160 @@
-import { run } from '@xata.io/shell';
+import { Flags } from '@oclif/core';
+import { XataApiClient } from '@xata.io/client';
+import { generateFromContext } from '@xata.io/codegen';
+import chalk from 'chalk';
+import EventEmitter from 'events';
+import { createWriteStream } from 'fs';
+import fs, { mkdir } from 'fs/promises';
+import fetch from 'node-fetch';
+import { homedir, tmpdir } from 'os';
+import path, { dirname } from 'path';
+import RJSON from 'relaxed-json';
+import repl from 'repl';
+import { fileURLToPath } from 'url';
 import { BaseCommand } from '../../base.js';
+import { readAPIKey } from '../../key.js';
+
+const __filename = fileURLToPath(import.meta.url);
+const __dirname = dirname(__filename);
 
 export default class Shell extends BaseCommand {
   static description = 'Open a shell to the current database and branch';
 
+  static flags = {
+    ...this.commonFlags,
+    code: Flags.string({
+      description: 'Fragment of code to be executed in the shell immediately after starting it'
+    })
+  };
+
   async run(): Promise<void> {
-    const databaseURL = await this.getDatabaseURL();
-    await run({ databaseURL });
+    const { flags } = await this.parse(Shell);
+    const apiKey = await readAPIKey();
+    if (!apiKey)
+      this.error('No API key found. Either use the XATA_API_KEY environment variable or run `xata auth login`');
+    const { protocol, host, databaseURL } = await this.getParsedDatabaseURL();
+
+    // Generate the file in the same dir than this package's code so it
+    // can import @xata.io/client
+    const tempFile = path.join(__dirname, `xata-${Date.now()}.mjs`);
+    try {
+      const { transpiled } = await generateFromContext('javascript', { apiKey, databaseURL });
```

Comment:
I believe ``generateFromContext`` should be deprecated in the ``codegen`` package, maybe not in this PR but in the near future.

Reply:
Yup. Let's do it in another PR.